presto-tests/src/test/java/com/facebook/presto/execution/TestHistoryBasedStatsTracking.java

```diff
@@ -471,6 +472,19 @@ public void testBroadcastJoin()
                         anyTree(any())));
     }
 
+    @Test
+    public void testFactPrioritization()
+    {
+        String query1 = "SELECT (SELECT nationkey FROM nation WHERE name = 'UNITED STATES') AS us_nationkey";
```

Comment:
does these test fail without your change?  i assume both HBO and CBO will report the result as one
 You could add a PlanMatcher to check additionally that the source and confidence levels of  the statistics and from CBO and are FACT statistics.

You could also consider artificially registering historical stats with a different row count and expect that the results are still one.